shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/database/metadata/dialect/PostgreSQLDataSourceMetaData.java

```diff
@@ -41,7 +41,7 @@
     
     private final String schema;
     
-    private final Pattern pattern = Pattern.compile("jdbc:postgresql://([\\w\\-\\.]+):?([0-9]*)/([\\w\\-]+)", Pattern.CASE_INSENSITIVE);
+    private final Pattern pattern = Pattern.compile("jdbc:postgresql://([\\w\\-\\.]+):?([0-9]*),?.*?/([\\w\\-]+)?\\S*", Pattern.CASE_INSENSITIVE);
```

Comment:
What is the different for use multiple hosts and replica-query rule?

We need to discuss the production design here. It is better to use same way for same feature. I prefer use replica-query rule to deal with all master-slave scenarios.

Reply:
Like this secene: There is PG clusters, contains one master node and another async slave node. 
The replica-query file can config as this:
**master : jdbc:postgresql://masterIp:5432,slaveIp:5432/db?targerServerType=master**
**slave : jdbc:postgresql://masterIp:5432,slaveIp:5432/db?targerServerType=preferSlave**
When PG master node goes down, I can promote async node change to new master fast. And the replica-query config file don't have to change. And the application don't have to restart. JDBC will recovery over a moment time.